tests/functional/simple_snapshot/test_check_cols_updated_at_snapshot.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+
+snapshot_sql = """
+{% snapshot snapshot_check_cols_updated_at_actual %}
+    {{
+        config(
+            target_database=database,
+            target_schema=schema,
+            unique_key='id',
+            strategy='check',
+            check_cols='all',
+            updated_at="'" ~ var("updated_at") ~  "'::timestamp",
+        )
+    }}
+
+    {% if var('version') == 1 %}
+
+        select 'a' as id, 10 as counter, '2016-01-01T00:00:00Z'::timestamp as timestamp_col union all
+        select 'b' as id, 20 as counter, '2016-01-01T00:00:00Z'::timestamp as timestamp_col
+
+    {% elif var('version') == 2 %}
+
+        select 'a' as id, 30 as counter, '2016-01-02T00:00:00Z'::timestamp as timestamp_col union all
+        select 'b' as id, 20 as counter, '2016-01-01T00:00:00Z'::timestamp as timestamp_col union all
+        select 'c' as id, 40 as counter, '2016-01-02T00:00:00Z'::timestamp as timestamp_col
+
+    {% else %}
+
+        select 'a' as id, 30 as counter, '2016-01-02T00:00:00Z'::timestamp as timestamp_col union all
+        select 'c' as id, 40 as counter, '2016-01-02T00:00:00Z'::timestamp as timestamp_col
+
+    {% endif %}
+
+{% endsnapshot %}
+"""
+
+snapshot__check_cols_updated_at_csv = """
+id,counter,timestamp_col,dbt_scd_id,dbt_updated_at,dbt_valid_from,dbt_valid_to
+a,10,2016-01-01 00:00:00.000,927354aa091feffd9437ead0bdae7ae1,2016-07-01 00:00:00.000,2016-07-01 00:00:00.000,2016-07-02 00:00:00.000
+b,20,2016-01-01 00:00:00.000,40ace4cbf8629f1720ec8a529ed76f8c,2016-07-01 00:00:00.000,2016-07-01 00:00:00.000,
+a,30,2016-01-02 00:00:00.000,e9133f2b302c50e36f43e770944cec9b,2016-07-02 00:00:00.000,2016-07-02 00:00:00.000,
+c,40,2016-01-02 00:00:00.000,09d33d35101e788c152f65d0530b6837,2016-07-02 00:00:00.000,2016-07-02 00:00:00.000,
+""".lstrip()
+
+
+@pytest.fixture(scope="class")
+def snapshots():
+    return {"my_snapshot.sql": snapshot_sql}
+
+
+@pytest.fixture(scope="class")
+def seeds():
+    return {"snapshot_check_cols_updated_at_expected.csv": snapshot__check_cols_updated_at_csv}
+
+
+@pytest.fixture(scope="class")
+def project_config_update():
+    return {
+        "seeds": {
+            "quote_columns": False,
+            "test": {
+                "snapshot_check_cols_updated_at_expected": {
+                    "+column_types": {
+                        "timestamp_col": "timestamp without time zone",
+                        "dbt_updated_at": "timestamp without time zone",
+                        "dbt_valid_from": "timestamp without time zone",
+                        "dbt_valid_to": "timestamp without time zone",
+                    },
+                },
+            },
+        },
+    }
+
+
+def test_simple_snapshot(project):
```

Comment:
Make sure I understand the test case correctly here.
`dbt seed` would create the`snapshot_check_cols_updated_at_expected ` table. All data in `snapshot_check_cols_updated_at_actual` is purely generated by the logic in the `snapshot.sql`. 

And for the command `run_dbt(["snapshot", "--vars", "{version: 3, updated_at: 2016-07-03}"])` command there's no data in snapshot table got generated because no data got modified.

This test is also designed to make sure the time for `dbt_updated_at` actually has the time that we passed in through '--vars'. Which is the main change this PR is doing

This is a great test! I would recommend adding a few short comments to document the intention of each command.

Reply:
Thank you for this excellent feedback @ChenyuLInx!

You are correct in your understanding of the test case. I've updated the description of the pull request to reflect your feedback. Or were you thinking off adding these as comments within the code itself?

Reply:
Added some comments to explain the testing approach.